lib/warren/framework_adaptor/rails_adaptor.rb

```diff
@@ -1,6 +1,35 @@
 # frozen_string_literal: true
 
 module Warren
+  # Namespace for framework adaptors.
+  # A FrameworkAdaptor should implement the following instance methods:
+  #
+  ## recovered? => Bool
```

Comment:
could these comments be above their respective methods?

Reply:
This is documentation for implementing a new framework adaptor (Say for sequel, rather than ActiveRecord) The actual RailsAdaptor methods are documented inline.